src/Dependencies.jl

```diff
@@ -113,43 +113,37 @@ end
 # Add JSON serialization of dependencies
 string_or_nothing(x) = isnothing(x) ? x : string(x)
 
-major(v::VersionNumber) = v.major
-minor(v::VersionNumber) = v.minor
-patch(v::VersionNumber) = v.patch
-major(v::Pkg.Types.VersionBound) = v.t[1]
-minor(v::Pkg.Types.VersionBound) = v.t[2]
-patch(v::Pkg.Types.VersionBound) = v.t[3]
-__version(v::VersionNumber) = v
-__version(v::Pkg.Types.VersionSpec) = v.ranges[1].lower
-version(d::AbstractDependency) = __version(getpkg(d).version)
-version(d::Dependency) = __version(d.pkg.version)
+# helper to get the version of a dependency (and no the build_version)
+version(d::AbstractDependency) = getpkg(d).version
+version(d::Dependency) = d.pkg.version
```

Comment:
What's the difference?  Isn't the first method sufficient?

Reply:
No it is (and was) not enough, which I tried to explain in the comment I added above

Reply:
Any suggestions on how to make this clearer are welcome :-)

Reply:
Just to make this explicit: this trick is not by me, it was added by @staticfloat as part of the build_version support; I merely dropped the (now unnecessary) `__version` and added a comment in an attempt to clarify what's going on (well, or at least give a hint as to what's going on). It can probably be phrased better (and I just noticed a typo).

Reply:
I was missing that `getpkg(d::Dependency)` isn't as simple as `d.pkg`.  At this point I find a bit questionable that method, since this is basically undoing it.